medusa/purge.py

```diff
@@ -62,6 +70,58 @@ def main(config, max_backup_age=0, max_backup_count=0):
         sys.exit(1)
 
 
+def find_backups_to_purge(
+        complete_cluster_backup_names, node_backups, prefer_incomplete, max_backup_age, max_backup_count
+):
+    backups_to_purge = set()
+
+    # we purge old backups always
+    backups_to_purge |= set(backups_to_purge_by_age(node_backups, max_backup_age))
+
+    # if we already keep only max_backup_count or less, we return
+    if max_backup_count == 0 or len(node_backups) - len(backups_to_purge) <= max_backup_count:
+        return backups_to_purge
+
+    # if we don't care about the incomplete ones, we purge by count and return
+    if not prefer_incomplete:
+        backups_to_purge |= set(backups_to_purge_by_count(node_backups, max_backup_count))
```

Comment:
**Issue:** Unless I'm missing something, I think this is doing the opposite of what was requested in the ticket.
If we don't prefer incomplete backups (which should mean we prefer complete backups?), then you're applying the same rules as before here, which can lead to having only incomplete backups left after the purge.
The implementation we're looking for here is to purge incomplete backups first (disregarding their age and count) and only then apply the age/count purge rules.

I think we could widely simplify this code and the rules by always purging incomplete backups in priority, and only then apply the purge rules.
wdyt?